viz/cmd/authz.go

```diff
@@ -87,104 +85,82 @@ func NewCmdAuthz() *cobra.Command {
-						route = row.GetSrvStats().GetRoute().Name
-					}
-					if row.Stats != nil {
-						rows = append(rows, table.Row{
-							route,
-							server,
-							authzp,
-							saz,
-							fmt.Sprintf("%.2f%%", getSuccessRate(row.Stats.GetSuccessCount(), row.Stats.GetFailureCount())*100),
-							fmt.Sprintf("%.1frps", getRequestRate(row.Stats.GetSuccessCount(), row.Stats.GetFailureCount(), row.TimeWindow)),
-							fmt.Sprintf("%dms", row.Stats.LatencyMsP50),
-							fmt.Sprintf("%dms", row.Stats.LatencyMsP95),
-							fmt.Sprintf("%dms", row.Stats.LatencyMsP99),
-						})
-					}
-				}
+			if options.allNamespaces && target.Name != "" {
+				return errors.New("stats for a resource cannot be retrieved by name across all namespaces")
+			}
 
-				// Unauthorized
-				requestParams = util.StatsSummaryRequestParams{
-					StatsBaseRequestParams: util.StatsBaseRequestParams{
-						TimeWindow:    options.timeWindow,
-						ResourceName:  server,
-						ResourceType:  k8s.Server,
-						Namespace:     options.namespace,
-						AllNamespaces: false,
-					},
-					ToNamespace:   options.namespace,
-					LabelSelector: options.labelSelector,
-				}
+			if options.allNamespaces {
+				target.Namespace = ""
+			} else if target.Namespace == "" {
+				target.Namespace = corev1.NamespaceDefault
+			}
 
-				req, err = util.BuildStatSummaryRequest(requestParams)
-				if err != nil {
-					return err
+			req.Resource = target
+
+			resp, err := client.Authz(cmd.Context(), &req)
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "Authz API error: %s", err)
+				os.Exit(1)
+			}
+			if e := resp.GetError(); e != nil {
+				fmt.Fprintf(os.Stderr, "Authz API error: %s", e.Error)
+				os.Exit(1)
+			}
+
+			for _, row := range resp.GetOk().GetStatTable().GetPodGroup().GetRows() {
+				server := row.GetSrvStats().GetSrv().GetName()
+				if row.GetSrvStats().GetSrv().GetType() == "default" {
+					server = fmt.Sprintf("%s:%s", row.GetSrvStats().GetSrv().GetType(), row.GetSrvStats().GetSrv().GetName())
 				}
-				resp, err = requestStatsFromAPI(client, req)
-				if err != nil {
-					fmt.Fprint(os.Stderr, err.Error())
-					os.Exit(1)
+				authz := fmt.Sprintf("%s/%s", row.GetSrvStats().GetAuthz().GetType(), row.GetSrvStats().GetAuthz().GetName())
+				if row.GetSrvStats().GetAuthz().GetType() == "" {
+					authz = ""
 				}
-				for _, row := range respToRows(resp) {
-					var route string
-					if row.GetSrvStats().GetRoute() != nil {
-						route = row.GetSrvStats().GetRoute().Name
-					}
-					if row.SrvStats != nil && row.SrvStats.DeniedCount > 0 {
-						rows = append(rows, table.Row{
-							route,
-							server,
-							unauthorized,
-							unauthorized,
-							"-",
-							fmt.Sprintf("%.1frps", getRequestRate(row.SrvStats.DeniedCount, 0, row.TimeWindow)),
-							"-",
-							"-",
-							"-",
-						})
+				if row.GetStats().GetSuccessCount()+row.GetStats().GetSuccessCount()+row.GetSrvStats().GetDeniedCount() > 0 {
```

Comment:
tinyy bit confused why we're adding the success count twice here.

Reply:
great question!  the answer is because this is a mistake 😂 it's supposed to be adding the successes, failures, and denials to determine if there are any metrics to display.